app/policies/proposal_policy.rb

```diff
@@ -83,25 +83,31 @@ def initialize(user, scope)
     end
 
     def resolve
-      # use subselects instead of left joins to avoid an explicit
-      # duplication-removal step
-      where_clause = <<-SQL
-        -- requester
-        requester_id = :user_id
-        -- approver / delegate
-        OR EXISTS (
-          SELECT * FROM approvals
-          LEFT JOIN approval_delegates ON (assigner_id = user_id)
-          WHERE proposal_id = proposals.id
-            -- TODO make visible to everyone involved
-            AND status <> 'pending'
-            AND (user_id = :user_id OR assignee_id = :user_id)
-        )
-        -- observer
-        OR EXISTS (SELECT id FROM observations
-                   WHERE proposal_id = proposals.id AND user_id = :user_id)
-        SQL
-      @scope.where(where_clause, user_id: @user.id)
+      if @user.app_admin?
+        @scope.all
+      elsif @user.admin?
+        @scope.where("client_data_type ilike ?", "#{@user.client_slug}%")
+      else
```

Comment:
Looks like you're changing the meaning of `admin?`. If so, mind changing all of the places it is called and replacing with `app_admin?`

Also, why is this a prefix match? Wouldn't an exact match be less error prone?